### PR TITLE
[Refactor] Cosmetic improvements to advantage modules

### PR DESCRIPTION
**Pull request by vmoens**
## Description

Some cosmetic improvements to GAE, TDL and TD.
Gathers the in_keys and out_keys automatically, and allows to use the modules without tensordict.

### CODE REVIEW

File: torchrl/objectives/value/advantages.py [hunk 1]
```diff
@@ -73,22 +88,64 @@ def is_functional(self):
...
+            >>> reward = torch.randn(1, 10, 1)
+            >>> done = torch.zeros(1, 10, 1, dtype=torch.bool)
+            >>> tensordict = TensorDict({"obs": obs, "next": {"obs": next_obs}, "done": done, "reward": reward}, [1, 10])
+            >>> _ = module(tensordict)
+            >>> assert "advantage" in tensordict.keys()
+
+        The module supports non-tensordict (i.e. unpacked tensordict) inputs too:
+
+        Examples:
+            >>> value_net = SafeModule(
```
Comment (vmoens) [inline torchrl/objectives/value/advantages.py:137]: cc @tcbegley this is something a bit more interesting than TensorDictModule
This module is a regular nn.Module that reads from a tensordict.
You may have a some crazy combination of keys that your value network needs to read, but you'd still be able to use regular tensors with this.
